dmutils/forms/dm_fields.py

```diff
@@ -0,0 +1,84 @@
+'''
+This module includes classes that should be used with WTForms within Digital Marketplace apps.
+
+They extend the classes found in `wtforms.fields`, and should be used instead of those.
+The advantage they provide is that they have been deliberately designed to be used with
+the Digital Marketplace frontend toolkit macros. If a particular field is missing, it
+should be added to this file.
+
+The main functionality is provided by the mixin class, `DMFieldMixin`. When a derived class
+includes `DMFieldMixin` in its base classes then the following extra features are provided:
+
+        - a `hint` property that can be set in the initialiser
+        - a `question` property that contains the label text
+        - a `value` property for the data that is displayed
+        - an `error` property for the field validation error
+
+For more details on how `DMFieldMixin`, see the documentation in `dmutils/forms/mixins.py`.
+'''
+
+import wtforms.fields
+
+import dmutils.forms.fields
+
+from .mixins import DMFieldMixin
+
+
+__all__ = ['DMBooleanField', 'DMDecimalField', 'DMEmailField', 'DMHiddenField',
+           'DMIntegerField', 'DMRadioField', 'DMStringField', 'DMStripWhitespaceStringField']
+
+
+class DMBooleanField(DMFieldMixin, wtforms.fields.BooleanField):
+    pass
+
+
+class DMDecimalField(DMFieldMixin, wtforms.fields.DecimalField):
+    pass
+
+
+class DMEmailField(DMFieldMixin, dmutils.forms.fields.EmailField):
+    pass
+
+
+class DMHiddenField(DMFieldMixin, wtforms.fields.HiddenField):
+    pass
+
+
+class DMIntegerField(DMFieldMixin, wtforms.fields.IntegerField):
+    pass
+
+
+class DMRadioField(DMFieldMixin, wtforms.fields.RadioField):
+    '''
+    A Digital Marketplace wrapper for `wtforms.RadioField`.
+
+    The `choices` argument for the constructor should be a sequence of
+    `(value, label, description)` tuples.
+
+    The `options` property is the choices in a format suitable for the
+    frontend toolkit.
+    '''
+    def __init__(self, label=None, validators=None, choices=None, **kwargs):
+        if choices:
+            # We do this the long way rather than using a comprehension because
+            # we want to be able to accept (value, label) tuples as well.
+            self.options = []
+            for choice in choices:
+                option = {}
+                option['value'] = choice[0]
+                option['label'] = choice[1]
+                if len(choice) > 2 and choice[2]:
+                    option['description'] = choice[2]
```

Comment:
It feels like we probably need to append the constructed `option` dict to `self.options`, unless I'm missing something magical. 🤔  I can't see where it would otherwise get persisted here.

Reply:
Fixed